manifests/init.pp

```diff
@@ -162,49 +195,88 @@
       additional_includes => ['/etc/pulp/vhosts80/*.conf'],
       custom_fragment     => template('foreman_proxy_content/httpd_pub.erb'),
     }
+  }
 
-    class { '::certs::qpid':
-      require => Class['certs'],
-    }
-    ~> class { '::qpid':
+  # current behaviour always install qpid with a pulp_node unless you explicitly disable local qpid
+  # pulp_master without standalone already has qpid through katello install
+  if $install_local_qpid and ( $pulp_node or $pulp_master_standalone ) {
+    include ::certs::qpid
+    class { '::qpid':
       ssl                    => true,
       ssl_cert_db            => $::certs::nss_db_dir,
       ssl_cert_password_file => $::certs::qpid::nss_db_password_file,
       ssl_cert_name          => 'broker',
       interface              => 'lo',
+      subscribe              => Class['::certs::qpid'],
     }
+  }
 
-    class { '::certs::qpid_client':
-      require => Class['certs'],
+  if $install_pulp {
+    include ::certs::qpid_client
+
+    if $manage_httpd {
+      $ca_cert = $::certs::ca_cert
+      $https_cert = $certs::apache::apache_cert
+      $https_key = $certs::apache::apache_key
+    } else {
+      $ca_cert = undef
+      $https_cert = undef
+      $https_key = undef
     }
-    ~> class { '::pulp':
-      enable_rpm                => true,
-      enable_puppet             => true,
-      enable_docker             => true,
-      enable_ostree             => $enable_ostree,
-      default_password          => $pulp_admin_password,
-      oauth_enabled             => true,
-      oauth_key                 => $pulp_oauth_key,
-      oauth_secret              => $pulp_oauth_secret,
-      messaging_transport       => 'qpid',
-      messaging_auth_enabled    => false,
-      messaging_ca_cert         => $certs::ca_cert,
-      messaging_client_cert     => $certs::messaging_client_cert,
-      messaging_url             => "ssl://${qpid_router_broker_addr}:${qpid_router_broker_port}",
+
+    if $pulp_node {
+      $node_oauth_effective_user = $pulp_oauth_effective_user
+      $node_oauth_key            = $pulp_oauth_key
+      $node_oauth_secret         = $pulp_oauth_secret
+      $node_server_ca_cert       = $certs::pulp_server_ca_cert
+    } else {
+      $node_oauth_effective_user = undef
+      $node_oauth_key            = undef
+      $node_oauth_secret         = undef
+      $node_server_ca_cert       = undef
+    }
+
+
+
+    class { '::pulp':
       broker_url                => "qpid://${qpid_router_broker_addr}:${qpid_router_broker_port}",
       broker_use_ssl            => true,
+      ca_cert                   => $ca_cert,
```

Comment:
This is currently unset for the "pulp master", see https://github.com/Katello/puppet-katello/blob/master/manifests/init.pp#L167-L197.
We need a setting to set this to `undef` on a standalone pulp master. This is currently not possible.